src/librustc/middle/const_eval.rs

```diff
@@ -416,6 +424,12 @@ pub enum ErrKind {
     ExpectedConstTuple,
     ExpectedConstStruct,
     TupleIndexOutOfBounds,
+    IndexedNonVec,
+    IndexNotNatural,
```

Comment:
I like `IndexNegative`, I'll go with that.

Natural numbers only include integers.